### PR TITLE
Add workflows for PRs and releases

### PR DESCRIPTION
**Pull request by ChrisLGardner**

### CODE REVIEW

**Comment by ChrisLGardner** [comment]
The release one isn't quite right yet, just need to figure out the right trigger for it.

**Comment by ChrisLGardner** [comment]
That should fix it now.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      run: go build -v ./... -ldflags="-s -w"
+      env:
+        GOOS: windows
+        GOARCH: amd64
+
+    - name: Build Windows
+      run: go build -v ./... -ldflags="-s -w"
+      env:
+        GOOS: linux
+        GOARCH: amd64
```
Comment (michaeltlombardi) [inline .github/workflows/release.yml]: It looks to me like these names are backwards?

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Looks like two of the jobs are mis-named - I think we also wanna build release binaries for MacOS

**Comment by ChrisLGardner** [comment]
Copy and paste got me there. I'll get that fixed and add Mac too. I'd left it off initially as I didn't see support for it in upx but might have overlooked it, though its probably worth building them anyway and letting people get them from github if needed.

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**